tool/src/cpp2/ty.rs

```diff
@@ -137,12 +137,9 @@ inline {ty_name}::~{ty_name}() {{
         let ty_name = self.cx.formatter.fmt_type_name(id);
         writeln!(self.decl_header, "struct {ty_name} {{").unwrap();
         for field in def.fields.iter() {
-            let decls = self.gen_ty_decl(&field.ty, field.name.as_str());
-            for (decl_ty, decl_name) in decls {
-                writeln!(self.decl_header, "\t{decl_ty} {decl_name};").unwrap();
-            }
+            let (decl_ty, decl_name) = self.gen_ty_decl(&field.ty, field.name.as_str());
```

Comment:
praise: ty_decl is a good name for this

i've been annoyed by the difference between a type "name" (for a custom type) and a type "name" (a way to name _any_ type, i.e a decl). It's been on my todo to figure out better terminology for C2 here. I think this works well!

Reply:
Thanks, though the name came from a similar function in C2 :)

Reply:
i guess we still have the ambiguity between type name and how you name a type.